### PR TITLE
Re-implement probe

### PR DESCRIPTION
**Pull request by dhodovsk**
This PR introduces probe with features:
- will run only for specified amount of time
- ignores expected exceptions
- accepts  outputs other than True

### CODE REVIEW

**Comment by dhodovsk** [comment]
Suggested fuctionality after discussion with @bkabrda 
- add posibility to specify infinite timeout
- define/fix behaviour when ```probe.run``` is called after ```probe.run_in_background```
- use multithreading instead of multiprocessing
- merge _ProbeHelper and Probe into one class

**Comment by jscotka** [comment]
Hi @dhodovsk, @bkabrda  why didn't you  use some library for this like library stolen from cockpit  ;-) (also inclueded in mtf)
https://github.com/cockpit-project/cockpit/blob/master/test/avocado/timeoutlib.py

it allows to use decorators and works like:
@Retry(attempts = 30, timeout = 30, error = Exception("too many"), delay = 1)
def do_something5():
   some_ip_probe()
do_something5()

**Comment by TomasTomecek** [comment]
@jscotka why didn't you publish that library on PyPI?

**Comment by jscotka** [comment]
@TomasTomecek  I'm not author, it was done by Milos, mentioned in conmment: 
https://github.com/cockpit-project/cockpit/blob/master/test/avocado/timeoutlib.py#L20

**Comment by jscotka** [comment]
otherwise there is more similar libnraries like: https://pypi.python.org/pypi/timeout-decorator
but when I've need that, it was not possbible to use it inside cockpit, because I need more functions, like inverse checks, and add some delays, to not do commands so quickly and much more, but iti possible that for this purpose, there could be also simplier decorator, than this from cockpit

File: conu/utils/probes.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        self.expected_exceptions = expected_exceptions
+        self.fnc = fnc
+        self.kwargs = kwargs
+        self.expected_retval = expected_retval
+        self.process = None
+        self.queue = None
+
+    def run(self):
+        if self.process and self.process.is_alive():
+            raise RuntimeError("One instance of probe can only be started once")
```
Comment (bkabrda) [inline conu/utils/probes.py]: Should the message rather say `One instance of Probe can only be probing once at any given time` or something like that?

File: conu/utils/probes.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        return self._run()
+
+    def run_in_background(self):
+        if self.process and self.process.is_alive():
+            raise RuntimeError("One instance of probe can only be started once")
```
Comment (bkabrda) [inline conu/utils/probes.py]: Same as above (message).

**Comment by bkabrda** [comment]
Just two minor nits, otherwise LGTM.

**Pull request merged**